doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/docUtils.kt

```diff
@@ -0,0 +1,404 @@
+            } else {
+                if (currentBlock.isEmpty()) {
+                    currentBlock = "$line\n"
+                } else {
+                    currentBlock += "$line\n"
+                }
+            }
+            var escapeNext = false
+            for (char in line) {
+
+                if (escapeNext) {
+                    escapeNext = false
+                    continue
+                }
+
+                when (char) {
+                    '\\' -> escapeNext = true
+
+                    '{' -> blocksIndicators += "{}"
+                    '}' -> blocksIndicators.removeAllElementsFromLast("{}")
+
+                    '[' -> blocksIndicators += "[]"
+                    ']' -> blocksIndicators.removeAllElementsFromLast("[]")
+
+                    '(' -> blocksIndicators += "()"
+                    ')' -> blocksIndicators.removeAllElementsFromLast("()")
+
+                    '<' -> blocksIndicators += "<>"
+                    '>' -> blocksIndicators.removeAllElementsFromLast("<>")
+
+                    '`' -> if (!blocksIndicators.removeAllElementsFromLast("`")) blocksIndicators += "`"
+
+                    // TODO html tags
+                }
+            }
+        }
+        add(currentBlock.removeSuffix("\n"))
+    }
+}
+
+/** Finds any inline tag with its depth, preferring the innermost one.
+ * "{@}" marks are ignored if "\" escaped.
+ */
+private fun DocContent.findInlineTagRangesWithDepthOrNull(): Pair<IntRange, Int>? {
+    var depth = 0
+    var start: Int? = null
+    var escapeNext = false
+    for ((i, char) in this.withIndex()) {
+        // escape this char
+        when {
+            escapeNext -> escapeNext = false
+
+            char == '\\' -> escapeNext = true
+
+            char == '{' && this.getOrNull(i + 1) == '@' -> {
+                start = i
+                depth++
+            }
+
+            char == '}' -> {
+                if (start != null) {
+                    return Pair(start..i, depth)
+                }
+            }
+        }
+    }
+    return null
+}
+
+/**
+ * Finds all inline tag names, including nested ones,
+ * together with their respective range in the doc.
+ * The list is sorted by depth, with the deepest tags first and then by order of appearance.
+ * "{@}" marks are ignored if "\" escaped.
+ */
+fun DocContent.findInlineTagNamesInDocContentWithRanges(): List<Pair<String, IntRange>> {
+    var text = this
+
+    return buildMap<Int, MutableList<Pair<String, IntRange>>> {
+        while (text.findInlineTagRangesWithDepthOrNull() != null) {
+            val (range, depth) = text.findInlineTagRangesWithDepthOrNull()!!
+            val comment = text.substring(range)
+            comment.getTagNameOrNull()?.let { tagName ->
+                getOrPut(depth) { mutableListOf() } += Pair(tagName, range)
+            }
+
+            text = text.replaceRange(
+                range = range,
+                replacement = comment
+                    .replace('{', '<')
```

Comment:
again, it repeats, yeah, better to extract to the constants with short and ASCII-like names

Reply:
YEAH, it's a tradeoff between common recommendation and readability